django/test/runner.py

```diff
@@ -581,6 +581,10 @@ def add_arguments(cls, parser):
             ),
         )
 
+    def log(self, msg, level=2):
```

Comment:
I had suggested on the ticket that this method accept a Python [logging level](https://docs.python.org/3/library/logging.html#logging-levels) like [`logging.log()`](https://docs.python.org/3/library/logging.html#logging.log), rather than an integer "verbosity." The logging level is the API that is most widely used and familiar to most people, and the goal of the ticket was to transition to using the logging module.

Looking at the [`--verbosity` docs](https://docs.djangoproject.com/en/3.1/ref/django-admin/#cmdoption-verbosity), I would say that a verbosity of 1 (the default) should show `INFO` and higher, while a verbosity of 2 or 3 (verbose or very verbose) should show `DEBUG`. The reason is that we want the ability to show some informational messages by default (like ticket #14137's shuffle seed). A verbosity of 0 should show no output. This logic can be reflected below.

Reply:
Sure. Correct me if I'm wrong, 
We define constants like 
```
# Log Levels
NOTSET = 0
DEBUG = 10
INFO = 20
...
```

And in the __init__ method of DiscoverRunner, instead of setting self.verbosity = verbosity, we set self.level depending on the value of verbosity.
```
def __init__(self, pattern=None, top_level=None, verbosity=1,
                 interactive=True, failfast=False, keepdb=False,
                 reverse=False, debug_mode=False, debug_sql=False, parallel=0,
                 tags=None, exclude_tags=None, test_name_patterns=None,
                 pdb=False, buffer=False, enable_faulthandler=True,
                 timing=False, **kwargs):

        self.pattern = pattern
        self.top_level = top_level
        self.top_level = top_level
        if verbosity == 0:
            self.level = NOTSET
        elif verbosity == 1:
            self.level = INFO
        else:
            self.level = DEBUG
        self.interactive = interactive
```

Reply:
> We define constants like

You don't need to define your own constants. You can use e.g. `logging.INFO`, etc. if you need to.

I would not modify `DiscoverRunner.__init__()`. Changes can be restricted to the new `DiscoverRunner.log()` using the current `self.verbosity` attribute.

Reply:
I came across multiple instances where constants where manually defined. 
https://github.com/django/django/blob/41e6b2a3c5e723256506b9ff49437d52a1f3bf43/django/core/checks/messages.py#L2-L6
https://github.com/django/django/blob/41e6b2a3c5e723256506b9ff49437d52a1f3bf43/django/contrib/messages/constants.py#L1-L5
Just out of curiosity, any specific reasons for that? I personally think using logging.DEBUG etc is the better approach.

I modified `self.verbosity` because it is not used anywhere else. Seemed redundant to keep it as a class member.

Reply:
> I modified self.verbosity because it is not used anywhere else. Seemed redundant to keep it as a class member.

I'm not sure what you mean. `self.verbosity` is used in a number of places in `DiscoverRunner`. `verbosity` is a documented argument, and subclassers (users of Django) can be using the attribute, too, so I think it should be kept.

Reply:
+1, keep self.verbosity because third-party packages will assume ```self.verbosity``` is part of DiscoverRunner. This is a breaking change that's out of the scope of the PR.